modules/heat_conduction/test/tests/ad_heat_conduction/tests

```diff
@@ -0,0 +1,13 @@
+[Tests]
+  [./stateful_jacobian]
+    type = 'PetscJacobianTester'
+    input = 'test.i'
+    ratio_tol = 1e-7
+    difference_tol = 1e-5
+    run_sim = True
+    petsc_version = '>=3.9'
+    requirement = 'AD heat conduction and the Jacobian shall be beautiful'
```

Comment:
Haha I like it

Reply:
Well, just borrowing from the words in test/tests/materials/ad_material/tests :)